### PR TITLE
A few small changes

### PR DESCRIPTION
**Pull request by adeschamps**

### CODE REVIEW

File: lcm/src/lcm/providers/udpm.rs [hunk 1]
```diff
@@ -181,6 +181,7 @@ impl<'a> UdpmProvider<'a> {
     pub fn unsubscribe(&mut self, subscription: Subscription) {
         self.subscriptions
             .retain(|&(ref sub, _)| *sub != subscription);
+        drop(subscription);
```
Comment (neachdainn) [inline lcm/src/lcm/providers/udpm.rs]: Why was this added? Subscription is already dropped at the end of the function (and it's really just an integer, so dropping doesn't really do anything).

I'm assuming this was Clippy recommended?

File: lcm/src/message.rs [hunk 2]
```diff
@@ -109,14 +109,14 @@ impl Marshall for String {
 
         let len = i32::decode(buffer)?;
         if len <= 0 {
-            return Err(DecodeError::invalid_size(len as i64));
+            return Err(DecodeError::invalid_size(i64::from(len)));
```
Comment (neachdainn) [inline lcm/src/message.rs:112]: This one is also an interesting change.

Comment (scottmcm) [reply lcm/src/message.rs:112]: This is suggested because `as` could truncate if `len` were to become larger or unsigned in future.  Using from makes the infallibility of the conversion obvious to the reader.

Comment (adeschamps) [reply lcm/src/message.rs:112]: That's correct.

File: lcm/src/lcm/providers/udpm.rs (hunk 1 above)
Comment (adeschamps) [reply lcm/src/lcm/providers/udpm.rs]: Since the function didn't consume `subscription` clippy recommended that the function take a reference instead. Calling drop to explicitly consume the variable silences the warning and ensures that the caller can't hold on to an invalid subscription.
